pkg/provider/cloud/vsphere/provider.go

```diff
@@ -253,7 +267,11 @@ func (v *VSphere) CleanUpCloudProvider(ctx context.Context, cluster *kubermaticv
 	}
 
 	if cluster.Spec.Cloud.VSphere.Tags != nil {
-		if err := syncDeletedClusterTags(ctx, restSession, cluster); err != nil {
+		// During cleanup, we only care about the side effect of deleting the tags from vSphere.
+		// We can safely ignore the returned list of managed tags because the cluster is being
+		// deleted, so there's no need to update its annotations.
```

Comment:
"so there's no need to update its annotations" We don't manage them in annotations anymore.